packages/infra/lib/shared/open-search-construct.ts

```diff
@@ -0,0 +1,101 @@
+import { CfnOutput, RemovalPolicy } from "aws-cdk-lib";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import {
+  CapacityConfig,
+  Domain,
+  EbsOptions,
+  EngineVersion,
+  IDomain,
+} from "aws-cdk-lib/aws-opensearchservice";
+import * as secret from "aws-cdk-lib/aws-secretsmanager";
+import { Construct } from "constructs";
+
+const masterUserName = "admin";
+
+export interface OpenSearchConstructProps {
+  region: string;
+  vpc: ec2.IVpc;
+  capacity: CapacityConfig;
+  ebs: EbsOptions;
+  encryptionAtRest?: boolean;
+}
+
+export default class OpenSearchConstruct extends Construct {
+  public readonly domain: IDomain;
+  public readonly creds: { username: string; secretName: string };
+
+  constructor(scope: Construct, id: string, props: OpenSearchConstructProps) {
+    super(scope, `${id}Construct`);
+
+    const { vpc, capacity, ebs, encryptionAtRest = true } = props;
+    const dataNodesCount = capacity.dataNodes ?? 1;
+
+    const secretName = "OpenSearchSecretName";
+    const credsSecret = new secret.Secret(this, secretName, {
+      secretName,
+      generateSecretString: { includeSpace: false },
+    });
+    this.creds = {
+      username: masterUserName,
+      secretName,
+    };
+
+    const osSg = new ec2.SecurityGroup(this, `${id}-sg`, {
+      securityGroupName: `${id}-sg`,
+      vpc,
+    });
+
+    const zoneAwareness =
+      dataNodesCount < 1
+        ? undefined
+        : dataNodesCount === 1
+        ? { enabled: false }
+        : {
+            enabled: true,
```

Comment:
nit: chained conditionals are difficult to follow, would suggest breaking this up into ifs

Reply:
I agree when there's more logic on the tertiary, in this case it seems cleaner than creating a variable (non-const) and having if/else.

I can change if you feel strongly about it, for sure, just think its NABD and actually cleaner/simpler.